services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/trial.rb

```diff
@@ -104,24 +109,11 @@ def retry_params = { llm_id:, llm_prompt_id:, dataset_id: }
           self.number = last_trial_number + 1
         end
 
-        private def query_llm
-          [].tap do |api_call_times|
-            test_examples.each do |test_example|
-              api_call_start_time = Time.zone.now
-              prompt = llm_prompt.prompt_with_student_response(test_example.student_response)
-              raw_text = llm.completion(prompt)
-              api_call_times << (Time.zone.now - api_call_start_time).round(2)
-
-              llm_feedback = LLMFeedbackResolver.run(raw_text:)
-              llm_assigned_status = LLMAssignedStatusResolver.run(raw_text:)
-
-              LLMExample.create!(trial: self, raw_text:, llm_feedback:, test_example:, llm_assigned_status:)
-            rescue => e
-              trial_errors << { error: e.message, test_example_id: test_example.id, raw_text: }.to_json
-              next
-            end
-            update_results(api_call_times:)
-          end
+        private def on_complete(_status, options)
+          trial = Trial.find(options['trial_id'])
+          trial.update!(trial_duration: Time.zone.now - Time.zone.parse(trial.trial_start_time))
+          CalculateResultsWorker.perform_async(options['trial_id'])
+          trial.trial_errors.empty? ? trial.update!(status: COMPLETED) : trial.update!(status: FAILED)
```

Comment:
If I'm reading this correctly, if there were 99 successful jobs and 1 errored job, this trial would be labeled `Failed` but the final results of the job would still be calculated at that moment. I think that's reasonable and I like that the final status is calculated.

In the future, we might consider a way to retry the errors (kinda like CircleCI's 'run from failed'). Probably over-engineering at this point, but if we have a lot of partial failures, we should consider it.

Reply:
Yeah.  I kind of debated on adding something like this in, but in general, the trials seem to be going through without errors so I punted on it.